Files/coupons/utils.py

```diff
@@ -0,0 +1,80 @@
+from flask import jsonify
+from Files import db
+from ..models import CouponsSchema, Coupons, CartSchema, Cart
+import random
+
+def generate_coupon_code():
+    while True:
+        res = ''.join(random.choice('0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ') for i in range(8))
+        coupon = db.session.query(Coupons).filter(Coupons.coupon_code==res).first()
+        coupon_schema=CouponsSchema()
+        output = coupon_schema.dump(coupon)
+        if not output:
+            return res
+
+def retrieve_all_coupons():
+    coupons_details = db.session.query(Coupons).all()
+    if not coupons_details:
+        return None
+    coupons_schema=CouponsSchema(many=True)
+    output = coupons_schema.dump(coupons_details)
+    return {"result":output}
+
+def retrieve_coupon_id(coupon_id):
+    coupon_details = db.session.query(Coupons).filter(Coupons.coupon_id==coupon_id).first()
+    if not coupon_details:
+        return None
+    coupons_schema=CouponsSchema()
+    output = coupons_schema.dump(coupon_details)
+    return {"result":output}
+
+def add_coupon(discount, limit, minimum_cart_value):
+    coupon_code = generate_coupon_code()
```

Comment:
Coupon code must also be taken as input. For example there can be a coupon "POW100" offering flat Rs. 100 discount. Or something related to bank offers and stuff. Code shouldn't be random.